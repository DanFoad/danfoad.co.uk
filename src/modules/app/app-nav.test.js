import React from 'react';
import { shallow } from 'enzyme';

import AppNav from './app-nav';

jest.mock('../../services/label-service');

describe('app-container', () => {
  const makeProps = extendProps => {
    return {
      transparentBackground: false,
      ...extendProps,
    };
  };

  let wrapper;
  let props;

  function renderContainer(extendProps) {
    props = makeProps(extendProps);
    wrapper = shallow(<AppNav {...props} />);
  }

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should match snapshot', () => {
    renderContainer();
    expect(wrapper).toMatchSnapshot();
  });

  describe('convertIndex', () => {
    it('should give numbers <9 a leading zero and increment', () => {
      renderContainer();
      const convertIndex = wrapper.instance().convertIndex;
      for (let i = 0; i < 9; i++) {
        expect(convertIndex(i)).toEqual(`0${i + 1}`);
      }
    });

    it("shouldn't give numbers >=9 a leading zero", () => {
      renderContainer();
      const convertIndex = wrapper.instance().convertIndex;
      for (let i = 9; i < 15; i++) {
        expect(convertIndex(i)).toEqual(i + 1);
      }
    });
  });
});
