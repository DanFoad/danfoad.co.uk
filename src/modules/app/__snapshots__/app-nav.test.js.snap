// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app-container should render correctly 1`] = `
<nav
  className="main-nav "
>
  <h1
    className="main-nav__title"
  >
    <a
      href="/"
    >
      translated general.name
      <span
        className="separator"
      >
         :: 
      </span>
      <span>
        translated general.role
      </span>
    </a>
    <div
      className="main-nav__edge"
    />
  </h1>
  <div
    className="main-nav__slant"
  />
  <div
    className="main-nav__links--desktop"
  >
    <ul
      className="main-nav__links"
    >
      <Link
        key="0"
        replace={false}
        to="/"
      >
        <li>
          <span
            className="index"
          >
            01
          </span>
          translated app.nav.links.home
        </li>
      </Link>
    </ul>
  </div>
  <div
    className="main-nav__links--mobile"
  >
    <div
      className="main-nav__links--menu"
      onClick={[MockFunction]}
    >
      <img
        alt=""
        src="/img/menu.png"
      />
    </div>
    <ul
      className="main-nav__links "
    >
      <Link
        key="0"
        replace={false}
        to="/"
      >
        <li>
          <span
            className="index"
          >
            01
          </span>
          translated app.nav.links.home
        </li>
      </Link>
    </ul>
  </div>
</nav>
`;

exports[`app-container should render correctly when expanded 1`] = `
<nav
  className="main-nav "
>
  <h1
    className="main-nav__title"
  >
    <a
      href="/"
    >
      translated general.name
      <span
        className="separator"
      >
         :: 
      </span>
      <span>
        translated general.role
      </span>
    </a>
    <div
      className="main-nav__edge"
    />
  </h1>
  <div
    className="main-nav__slant"
  />
  <div
    className="main-nav__links--desktop"
  >
    <ul
      className="main-nav__links"
    >
      <Link
        key="0"
        replace={false}
        to="/"
      >
        <li>
          <span
            className="index"
          >
            01
          </span>
          translated app.nav.links.home
        </li>
      </Link>
    </ul>
  </div>
  <div
    className="main-nav__links--mobile"
  >
    <div
      className="main-nav__links--menu"
      onClick={[MockFunction]}
    >
      <img
        alt=""
        src="/img/menu.png"
      />
    </div>
    <div
      className="main-nav__links--overlay"
      onClick={[MockFunction]}
    />
    <ul
      className="main-nav__links open"
    >
      <Link
        key="0"
        replace={false}
        to="/"
      >
        <li>
          <span
            className="index"
          >
            01
          </span>
          translated app.nav.links.home
        </li>
      </Link>
    </ul>
  </div>
</nav>
`;

exports[`app-container should render correctly with a transparent background 1`] = `
<nav
  className="main-nav main-nav--transparent"
>
  <h1
    className="main-nav__title"
  >
    <a
      href="/"
    >
      translated general.name
      <span
        className="separator"
      >
         :: 
      </span>
      <span>
        translated general.role
      </span>
    </a>
    <div
      className="main-nav__edge"
    />
  </h1>
  <div
    className="main-nav__slant"
  />
  <div
    className="main-nav__links--desktop"
  >
    <ul
      className="main-nav__links"
    >
      <Link
        key="0"
        replace={false}
        to="/"
      >
        <li>
          <span
            className="index"
          >
            01
          </span>
          translated app.nav.links.home
        </li>
      </Link>
    </ul>
  </div>
  <div
    className="main-nav__links--mobile"
  >
    <div
      className="main-nav__links--menu"
      onClick={[MockFunction]}
    >
      <img
        alt=""
        src="/img/menu.png"
      />
    </div>
    <ul
      className="main-nav__links "
    >
      <Link
        key="0"
        replace={false}
        to="/"
      >
        <li>
          <span
            className="index"
          >
            01
          </span>
          translated app.nav.links.home
        </li>
      </Link>
    </ul>
  </div>
</nav>
`;
